lucene/core/src/java/org/apache/lucene/codecs/blocktree/SegmentTermsEnumFrame.java

```diff
@@ -163,15 +179,48 @@ void loadBlock() throws IOException {
     // instead of linear scan to find target term; eg
     // we could have simple array of offsets
 
+    final long startSuffixFP = ste.in.getFilePointer();
     // term suffixes:
-    code = ste.in.readVInt();
-    isLeafBlock = (code & 1) != 0;
-    int numBytes = code >>> 1;
-    if (suffixBytes.length < numBytes) {
-      suffixBytes = new byte[ArrayUtil.oversize(numBytes, 1)];
+    if (version >= BlockTreeTermsReader.VERSION_COMPRESSED_SUFFIXES) {
+      final long codeL = ste.in.readVLong();
+      isLeafBlock = (codeL & 0x04) != 0;
+      final int numSuffixBytes = (int) (codeL >>> 3);
+      if (suffixBytes.length < numSuffixBytes) {
+        suffixBytes = new byte[ArrayUtil.oversize(numSuffixBytes, 1)];
+      }
+      compressionAlg = (int) codeL & 0x03;
```

Comment:
An enum with final code field for clarity in this and other places, maybe? Should be effectively the same at runtime.

Reply:
I pushed a change that introduces an enum, does it look better?

Reply:
Even fancier than I though it'd have to be! Looks great I think.